src/sentry/dynamic_sampling/latest_release_booster.py

```diff
@@ -0,0 +1,72 @@
+from datetime import datetime
+
+from django.conf import settings
+from pytz import UTC
+
+from sentry.utils import redis
+
+BOOSTED_RELEASE_TIMEOUT = 60 * 60
+ONE_DAY_TIMEOUT = 60 * 60 * 24
+
+
+def get_redis_client_for_ds():
+    cluster_key = getattr(settings, "SENTRY_DYNAMIC_SAMPLING_RULES_REDIS_CLUSTER", "default")
+    return redis.redis_clusters.get(cluster_key)
+
+
+def generate_dynamic_sampling_lock_key(project_id):
+    return f"dynamic-sampling-boost-lock:{project_id}"
+
+
+def generate_cache_key_for_observed_release(project_id, release_id):
+    """
+    Generates a cache key for releases that had a transaction observed in the last 24 hours
+    """
+    return f"ds::p:{project_id}:r:{release_id}"
+
+
+def generate_cache_key_for_boosted_release(project_id):
+    """
+    Generates a cache key for the boosted releases for a given project.
+    """
+    return f"ds::p:{project_id}:boosted_releases"
+
+
+def observe_release_in_last_24h(project_id, release_id):
```

Comment:
_nit_: The return value of this function is not clear. Could we make this a boolean or something indicative of whether the caller should take action? For instance, by returning `release_observed_in_last_24h != "1"`